src/ipa_pc/mod.rs

```diff
@@ -773,8 +737,7 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
                     combined_rand += &(cur_challenge * &shift * &randomness.rand[segments_count - 1]);
                 }
 
-                cur_challenge = opening_challenges(opening_challenge_counter);
-                opening_challenge_counter += 1;
+                cur_challenge = fs_rng.squeeze_128_bits_challenge();
```

Comment:
Let's comment here, that this is the variant not targeting recursion. 
In this setting squeezing a fresh challenge is cheaper than doing a field multiplication.

Reply:
I am not sure if we want to distinguish between recursion-friendly and not recursion-friendly implementation at all?
If not, I'd recommend to implement the recursion friendly variant only, which takes powers of the challenge.

Reply:
Maybe let's stick that, with this release, we need to support only final proofs. In the future we can make this distinction according to our needs. Let's put the comment as Ulrich suggested anyway

Reply:
Done: switched to powers